docs/examples/fork-process-model/README.rst

```diff
@@ -0,0 +1,51 @@
+Working With Fork Process Models
+================================
+
+The `BatchExportSpanProcessor` is not fork-safe and doesn't work well with application servers
+(Gunicorn, uWSGI) which are based on the pre-fork web server model. The `BatchExportSpanProcessor`
+spawns a thread to run in the background to export spans to the telemetry backend. During the fork, the child
+process inherits the lock which is held by the parent process and deadlock occurs. We can use fork hooks to
+get around this limitation of the span processor.
+
+Please see http://bugs.python.org/issue6721 for the problems about Python locks in (multi)threaded
+context with fork.
+
+Gunicorn post_fork hook
+-----------------------
+
+.. code-block:: python
+
+    from opentelemetry import trace
+    from opentelemetry.exporter.jaeger import JaegerSpanExporter
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+
+    def post_fork(server, worker):
+        server.log.info("Worker spawned (pid: %s)", worker.pid)
+        trace.set_tracer_provider(TracerProvider())
+        trace.get_tracer_provider().add_span_processor(
+            BatchExportSpanProcessor(JaegerSpanExporter(service_name='my-service'))
+        )
+
+
+uWSGI postfork decorator
+------------------------
+
+.. code-block:: python
+
+    from uwsgidecorators import postfork
+
+    from opentelemetry import trace
+    from opentelemetry.exporter.jaeger import JaegerSpanExporter
+    from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+
+    @postfork
+    def set_span_processor():
+        trace.get_tracer_provider().add_span_processor(
+            BatchExportSpanProcessor(JaegerSpanExporter(service_name='my-service'))
```

Comment:
Shouldn't we have a full example here that sets up the entire tracing pipeline including setting the tracer provider like we did in gunicorn example above?

Reply:
We can do that but one problem with setting tracer provider in the postfork is if there is `get_tracer` call somewhere in the sub modules then provider is set `DefaultTracerProvider` and overriding is not allowed in postfork which leads to errors. As I mentioned here https://github.com/open-telemetry/opentelemetry-python/pull/1609#discussion_r575796075 it's little hacky and mess right now.

Reply:
I mean let's show the full example of whatever is the recommended way to setup with uWSGI. This example doesn't show how/where to setup tracer provider. I think it should. If there are multiple ways and all of them have caveats then we mention both ways and the caveats the come with.